### ISSUE TITLE
[Help] <title>多任务p-tuning经验请教

### ISSUE THREAD

**Issue by YefZhao** [THUDM/ChatGLM-6B#1056]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

使用p-tuning无量化微调GLM-6B，多任务学习混合训练（特殊领域NER、意图识别、文本纠错等6个任务），每个任务使用不同的输入模板。目前模型效果能基本识别不同的任务输入，但每个任务效果不佳，特别是特殊领域NER，表现出知识学习不足（知道这是个文本纠错的任务，但改不对），准确率仅13%，同样训练量的t5模型达40%。
以下是一些参数
数据量共41M条；32batch size per device、8卡；learning rate：1e-2；linear_lr_scheduler（默认）；pre_seq_len：128（demo默认）；训练了0.8个epoch。

### Expected Behavior

目标：能更好地记住每个任务特殊的“知识”，提高每个任务上的表现。

一些疑问：
1. 模型能很好的区分6个任务，但每个任务的效果都一般，是否需要增大pre_seq_len（p-tuning所训练的那组参数）以拥有更大的知识存储空间；
2. 是否需要在使用linear_lr_scheduler的基础上加上warmup和weight decay；
3. 是否需要多轮训练以加强模型对知识的掌握；
4. 如有其他关于多任务p-tuning的经验，烦请多多指教。

### Steps To Reproduce

1. 将每个任务的数据同一为input、output的格式，不同任务的输入有自己统一的任务描述。
2. 所有41M条训练数据打乱，一起训练0.8epoch。
3. 测试模型在不同任务上的表现：能区分不同的任务，但各任务效果不佳。

### Environment

```markdown
OS: Ubuntu 20.04
Python: 3.8
Transformers: 4.26.1
PyTorch: 1.12
CUDA Support: True
```

**Comment by crowningwang** [comment]
您好，我在全量微调的时候，数据集仅加载了0.7M就报错RuntimeError: [4] is setting up NCCL communicator and retrieving ncclUniqueId from [0] via c10d key-value store by key '0', but store->get('0') got error: Socket Timeout，请问您遇到过吗

**Comment by YefZhao** [comment]
> 您好，我在全量微调的时候，数据集仅加载了0.7M就报错RuntimeError: [4] is setting up NCCL communicator and retrieving ncclUniqueId from [0] via c10d key-value store by key '0', but store->get('0') got error: Socket Timeout，请问您遇到过吗

您好，我没做全量微调，不过看报错像是网络问题。
